Lib/test/test_typing.py

```diff
@@ -5355,6 +5355,23 @@ class C(Generic[T]): pass
         self.assertEqual(get_args((*tuple[*Ts],)[0]), (*Ts,))
         self.assertEqual(get_args(Unpack[tuple[Unpack[Ts]]]), (tuple[Unpack[Ts]],))
 
+    def test_get_orig_class(self):
+        T = TypeVar('T')
+        class C(Generic[T]): pass
+        self.assertEqual(get_orig_class(C[int]()), C[int])
+        self.assertIsNone(get_orig_class(list[int]()))
+        self.assertIsNone(get_orig_class(int))
+
+    def test_get_orig_bases(self):
+        T = TypeVar('T')
+        class C(Generic[T]): pass
```

Comment:
Let's add `class A: ...` with no bases and test it. I would also recommend `class B(object)` (since it is a special case).